zerver/models.py

```diff
@@ -214,8 +215,8 @@ def remote_user_to_email(remote_user):
 
 class RealmEmoji(models.Model):
     realm = models.ForeignKey(Realm)
-    name = models.TextField()
-    img_url = models.TextField()
+    name = models.TextField(validators=[MinLengthValidator(1), RegexValidator(regex=r'^[0-9a-zA-Z_\-\.]*$')])
```

Comment:
OK, probably makes sense to harden the regex then.  Ideally we wouldn't accept names that are going to cause problems with the frontend (though arguably that should be fixed by encoding the name properly?).